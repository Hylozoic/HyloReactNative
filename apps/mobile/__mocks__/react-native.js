import React, {useState} from 'react'
import * as ReactNative from 'react-native'

export const Switch = props => {
  const [value, setValue] = useState(props.value)
  return (
    <ReactNative.TouchableOpacity
      onPress={() => {
        props.onValueChange(!value)
        setValue(!value)
      }}
      testID={props.testID}>
      <ReactNative.Text>{value?.toString()}</ReactNative.Text>
    </ReactNative.TouchableOpacity>
  )
}

newRN = Object.defineProperty(ReactNative, 'Switch', {
  get: function() {
    return Switch
  }
})

module.exports = newRN
