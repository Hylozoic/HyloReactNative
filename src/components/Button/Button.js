import React from 'react'
import { Text, View, TouchableOpacity } from 'react-native'
import { caribbeanGreen } from 'style/colors'
import Icon from '../Icon'
import { omit } from 'lodash/fp'

export default function Button ({
  style = {},
  text,
  onPress,
  iconName,
  disabled
 }) {
  const {
    color = 'white',
    borderColor = 'white',
    fontSize = 13,
    height = 30
  } = style

  const containerStyle = {
    ...styles.container,
    ...omit([
      'color', 'backgroundColor', 'disabledBackgroundColor', 'fontSize', 'height'
    ], style)}

  const backgroundColor = (disabled
    ? style.disabledBackgroundColor
    : style.backgroundColor) || caribbeanGreen

  const buttonStyle = {...styles.button, backgroundColor, height, borderColor}
  const textStyle = {...styles.text, color, fontSize}

  return <View style={containerStyle}>
    <TouchableOpacity disabled={disabled} onPress={disabled ? () => {} : onPress} style={styles.wrapper}>
      <View style={buttonStyle}>
        <View style={styles.buttonInner}>
          {iconName && <Icon name={iconName} style={styles.icon} />}
          <Text style={textStyle}>{text}</Text>
        </View>
      </View>
    </TouchableOpacity>
  </View>
}

const styles = {
  container: {
    alignItems: 'center'
  },
  wrapper: {
    flexDirection: 'row'
  },
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
    borderRadius: 100,
    borderWidth: 1
  },
  buttonInner: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'row'
  },
  text: {
    fontFamily: 'Circular-Bold'
  },
  icon: {
    color: 'white',
    fontSize: 16,
    paddingRight: 5
  }
}
