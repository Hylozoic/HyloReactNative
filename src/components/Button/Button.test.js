import 'react-native'
import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import Button from './Button'

it('matches last snapshot', () => {
  const renderer = new ReactShallowRenderer()
  const props = {
    text: 'Push Me',
    onPress: () => {},
    disabled: false
  }

  renderer.render(<Button {...props} />)
  const actual = renderer.getRenderOutput()

  expect(actual).toMatchSnapshot()
})

it('disables', () => {
  const renderer = new ReactShallowRenderer()
  const props = {
    text: 'Push Me',
    onPress: () => {},
    disabled: true
  }

  renderer.render(<Button {...props} />)
  const actual = renderer.getRenderOutput()

  expect(actual).toMatchSnapshot()
})
