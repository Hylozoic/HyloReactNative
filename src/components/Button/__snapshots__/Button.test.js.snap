// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disables 1`] = `
<View>
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={true}
    onPress={[Function]}
  >
    <View>
      <View>
        <Text>
          Push Me
        </Text>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`matches last snapshot 1`] = `
<View>
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={false}
    onPress={[Function]}
  >
    <View>
      <View>
        <Text>
          Push Me
        </Text>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
